src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -23,7 +23,7 @@ public class JsonRuleEngine : IRuleEngine
         public delegate ILineNumberResolver BuildILineNumberResolver(TemplateContext context);
 
         private readonly BuildILineNumberResolver BuildLineNumberResolver;
-        internal readonly IReadOnlyList<RuleDefinition> RuleDefinitions;
+        internal IList<RuleDefinition> RuleDefinitions;
```

Comment:
Keep the type `IReadOnlyList` - you can still assign a new value to it, but it will prevent changes to *List* itself.